build.gradle

```diff
@@ -109,8 +118,8 @@ if (project.hasProperty('deployment')) {
         def snapshotsRepoUrl = deployment.snapshot.url
         url = version.endsWith('SNAPSHOT') ? snapshotsRepoUrl : releasesRepoUrl
         credentials {
-          username = version.endsWith('SNAPSHOT') ? deployment.snapshot.username : deployment.release.username
-          password = version.endsWith('SNAPSHOT') ? deployment.snapshot.password : deployment.release.password
+          username = System.getenv('SONATYPE_USERNAME')
+          password = System.getenv('SONATYPE_PASSWORD')
```

Comment:
Is it possible for any person to create a PR, print this values and see them on the CircleCI UI?

Reply:
Great question. A few things are in play that should keep these values safe.

1. The variables are only provided via a CircleCI [context](https://circleci.com/docs/2.0/contexts/#overview). The 'context' in use is [restricted](https://circleci.com/docs/2.0/contexts/#restricting-a-context). Basically, not everyone will be able to run a build that uses this 'context'.

1. CI is configured to only run 'release' builds (which require access to the restricted 'context') on the 'master' branch.

1. As a secondary safeguards, CircleCI will do [secrets masking](https://circleci.com/docs/2.0/contexts/#secrets-masking).

So...I think it is safe, but if not, we'll fix it.